src/Authentication/Authentication/Helpers/ContentHelper.cs

```diff
@@ -0,0 +1,313 @@
+            if (!string.IsNullOrEmpty(protocol))
+            {
+                var statusCode = WebResponseHelper.GetStatusCode(response);
+                var statusDescription = WebResponseHelper.GetStatusDescription(response);
+                raw.AppendFormat("{0} {1} {2}", protocol, statusCode, statusDescription);
+                raw.AppendLine();
+            }
+
+            HttpHeaders[] headerCollections =
+            {
+                response.Headers,
+                response.Content == null ? null : response.Content.Headers
+            };
+
+            foreach (var headerCollection in headerCollections)
+            {
+                if (headerCollection == null)
+                {
+                    continue;
+                }
+
+                foreach (var header in headerCollection)
+                {
+                    // Headers may have multiple entries with different values
+                    foreach (var headerValue in header.Value)
+                    {
+                        raw.Append(header.Key);
+                        raw.Append(": ");
+                        raw.Append(headerValue);
+                        raw.AppendLine();
+                    }
+                }
+            }
+
+            raw.AppendLine();
+            return raw;
+        }
+
+        internal static bool IsJson(string contentType)
+        {
+            contentType = GetContentTypeSignature(contentType);
+            return CheckIsJson(contentType);
+        }
+
+        internal static bool IsText(string contentType)
+        {
+            contentType = GetContentTypeSignature(contentType);
+            return CheckIsText(contentType);
+        }
+
+        internal static bool IsXml(string contentType)
+        {
+            contentType = GetContentTypeSignature(contentType);
+            return CheckIsXml(contentType);
+        }
+
+        #endregion Internal Methods
+
+        #region Private Helper Methods
+
+        private static bool CheckIsJson(string contentType)
+        {
+            if (string.IsNullOrEmpty(contentType))
+                return false;
+
+            // the correct type for JSON content, as specified in RFC 4627
+            var isJson = contentType.Equals("application/json", StringComparison.OrdinalIgnoreCase);
+
+            // add in these other "javascript" related types that
+            // sometimes get sent down as the mime type for JSON content
+            isJson |= contentType.Equals("text/json", StringComparison.OrdinalIgnoreCase)
+                      || contentType.Equals("application/x-javascript", StringComparison.OrdinalIgnoreCase)
+                      || contentType.Equals("text/x-javascript", StringComparison.OrdinalIgnoreCase)
+                      || contentType.Equals("application/javascript", StringComparison.OrdinalIgnoreCase)
+                      || contentType.Equals("text/javascript", StringComparison.OrdinalIgnoreCase);
```

Comment:
to avoid repeating `StringComparison.OrdinalIgnoreCase`, you can create a `switch` block on `contentType.ToLower(CultureInfo.InvariantCulture)`